pilot/pkg/proxy/envoy/v2/ads.go

```diff
@@ -517,7 +537,13 @@ func (s *DiscoveryServer) pushAll(con *XdsConnection, pushEv *XdsEvent) error {
 				time.Since(pushEv.push.Start), string(out))
 		}
 	}()
-	// TODO: check version, suppress if changed
+	// check version, suppress if changed.
+	currentVersion := versionInfo()
+	if suppressPush && pushEv.version != currentVersion {
```

Comment:
Pretty sure now that this is !allowConcurrentPush and that the env var you have above needs to be inverted. Ie false by default.